creativecommons/cccatalog
# Temporary Flickr bugfix:  Reduce time interval requested from one day to 30 minutes

## Fixes


Related to #326  by @mathemancer 

## Description


This is a partial fix for the bug described in #326.  It reduces the amount of data requested at once by taking half hour intervals rather than the whole day at once when the script is run for a given date.

## Technical details


This fix is *incomplete*.  It's not clear why the Flickr API returns different result sets for different time interval sizes.  More investigation is necessary to come up with a complete solution.  This temporary fix is just to allow us to keep ingesting Flickr data while determining a better solution.

## Tests


If you have a Flickr API key available, you can do the following:
- start the dev environment using the `README.md`.
- run the tests (obviously)
- modify the `.env` file to contain your Flickr API key.
- restart the webserver container `docker-compose up  --no-deps webserver`
- in the container, in the `dags/provider_api_scripts` directory, run `python flickr.py --date 2018-08-20`.  You should get 114,000-115,000 rows in the resulting TSV file (you can see the logs using the interface by navigating your browser to `localhost:9090`, and about 83,000-84,000 unique rows in the DB (to check, use `psql -h localhost -U deploy -d openledger`, count the rows in the `image` table with `SELECT COUNT(*) FROM image;`)
- Delete the data from the table `DELETE FROM image; VACUUM image;` (from within `psql` interface still).
- Check out master, and run `python flickr.py --date 2018-08-20` in the container again (same directory; no need to restart any containers).  You should only get 4,500-5,000 unique rows in the DB this time (regardless of what the logs say).

This shows the bug in master, and the fact that the fix means we get around 17-20x more unique images' data per day.

## Checklist
 
  index.md`).
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
